### PR TITLE
update rule for mocking api responses

### PR DESCRIPTION
**Pull request by william-liebenberg**

### CODE REVIEW

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 1]
```diff
@@ -1,27 +1,132 @@
...
+- title: William Liebenberg
+  url: https://ssw.com.au/people/william-liebenberg
 - title: Bryden Oliver
   url: https://ssw.com.au/people/bryden-oliver
 related: 
 - do-you-know-how-to-brand-your-api-portal
 - do-you-know-the-best-tools-to-manage-apis
 redirects:
-- comments-do-you-know-when-to-mockup-your-apis
+- do-you-know-when-to-mockup-your-apis
```
Comment (bradystroud) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:17]: The redirects list should have all the rule's old URI's.
Please change it back to 
```yml
redirects:
- comments-do-you-know-when-to-mockup-your-apis
```

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 2]
```diff
@@ -1,27 +1,132 @@
...
+To complete the following steps, you need to create an instance of Azure API Management in your subscription.
+
+> TIP: If you have Azure Credits, the Developer (NO SLA) edition is the cheapest option to test most of the features offered by API Management.
+
+### Step 1 - Create Blank API
+
+1. Select **Blank API**
+
+![](step1-create-blank-api.png)
+Figure: Create a blank API
```
Comment (bradystroud) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:44]: The Figure statement should be put in the alt text
Change from 
```md
![](step1-create-blank-api.png)
Figure: Create a blank API
```
to
```md
![Figure: Create a blank API](step1-create-blank-api.png)
```

see the example rule as a guide https://github.com/SSWConsulting/SSW.Rules.Content/edit/main/rules/rule/rule.md

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 2 above)
Comment (pierssinclairssw) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:44]: The figure should go inside the square brackets, the text won't show up on GitHub previews but will when deployed.

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 3]
```diff
@@ -1,27 +1,132 @@
...
+
+## Step 2 - Describe your API
+
+![](step2-describe-your-api.png)
+Figure: Describe your API
```
Comment (pierssinclairssw) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:49]: The figure should go inside the square brackets, the text won't show up on GitHub previews but will when deployed.

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 4]
```diff
@@ -1,27 +1,132 @@
...
+1. Click **Create**
+
+## Step 3 - Add an Operation
+
+1. Select your API from the API list
+2. Click **Add operation**
+3. Specify the Display name, Name and URL of your operation 
+
+![](step3-add-operation.png)
+Figure: Adding a new Operation
```
Comment (pierssinclairssw) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:60]: The figure should go inside the square brackets, the text won't show up on GitHub previews but will when deployed.

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 5]
```diff
@@ -1,27 +1,132 @@
...
+
+## Step 4 - Add response
+
+1. Click **Add response**
+2. Select the appropriate response code (e.g 200 OK)
+
+![](step4-add-response.png)
+Figure: Add a response
```
Comment (pierssinclairssw) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:68]: The figure should go inside the square brackets, the text won't show up on GitHub previews but will when deployed.

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 6]
```diff
@@ -1,27 +1,132 @@
...
+
+1. Specify the **Content Type** as `application/json`
+2. Specify the **Sample** as `{ "data": "fake" }`
+3. Click **Save**
+  
+![](step5-add-represtntation.png)
+Figure: Add a representation for your response with fake data
+
+![](step5-operation-done.png)
+Figure: Operation added successfully
```
Comment (pierssinclairssw) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:80]: The figure should go inside the square brackets, the text won't show up on GitHub previews but will when deployed.

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 7]
```diff
@@ -1,27 +1,132 @@
...
+
+## Step 6 - Add Mock Response Policy
+
+1. Click **Add policy**
+2. Select **Mock responses**
+
+![](step6-add-mock-response-policy.png)
+Figure: Add the Mock Response policy
```
Comment (pierssinclairssw) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:88]: The figure should go inside the square brackets, the text won't show up on GitHub previews but will when deployed.

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 8]
```diff
@@ -1,27 +1,132 @@
...
+
+## Step 7 - Choose response
+
+1. Choose the response representation to return as a Mock (e.g. `200 OK, application/json`)
+2. Click **Save**
+
+![](step7-choose-response-representation.png)
+Figure: Choose the response representation
```
Comment (pierssinclairssw) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:96]: The figure should go inside the square brackets, the text won't show up on GitHub previews but will when deployed.

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 9]
```diff
@@ -1,27 +1,132 @@
...
+## Step 8 - Test your Mocked API
+
+1. Make sure your API is selected
+2. Click the **Test** tab
+3. Click the FakeData operation
+4. Note the "Mocking is enabled" notice
+5. Click **Send**
+
+![](step8-test-mock-api.png)
+Figure: Testing the mocked api operation
```
Comment (pierssinclairssw) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:107]: The figure should go inside the square brackets, the text won't show up on GitHub previews but will when deployed.

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 10]
```diff
@@ -1,27 +1,132 @@
...
+
+## Step 9 - Inspect Mocked API response
+
+1. Notice the 200 OK status code
+2. Notice the fake data response in the message body
+
+![](step9-inspect-mocked-api-reponse.png)
+Figure: Response from the mocked operation
```
Comment (pierssinclairssw) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:115]: The figure should go inside the square brackets, the text won't show up on GitHub previews but will when deployed.

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 11]
```diff
@@ -1,27 +1,132 @@
...
+1. Add a new `GET` request to your REST Client
+2. Specify the API Uri such as `https://william-ssw.azure-api.net/data`
+3. Add the subscription header `Ocp-Apim-Subscription-Key` and value
+     - you can copy the value directly from the operation HTTP Request section
+     - or, from the API Management service | Subscriptions blade
+4. **SEND** your request
+5. Verify the `200 OK` response and the mocked data
+
+![](step10-call-from-rest-client.png)
+Figure: Calling our Data API from a REST Client
```
Comment (pierssinclairssw) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:128]: The figure should go inside the square brackets, the text won't show up on GitHub previews but will when deployed.

**Review by pierssinclairssw — CHANGES_REQUESTED** [review]

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 12]
```diff
@@ -1,27 +1,132 @@
...
+![](step2-describe-your-api.png)
+Figure: Describe your API
+
+1. Click **Create**
+
+## Step 3 - Add an Operation
+
+1. Select your API from the API list
+2. Click **Add operation**
+3. Specify the Display name, Name and URL of your operation
```
Comment (pierssinclairssw) [inline rules/do-you-know-when-to-mockup-your-api/rule.md:57]: What is the difference between Display name and Name?

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 1 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md:17]: done

[the remark this answers is not in the document]
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md:17]: done

File: rules/do-you-know-when-to-mockup-your-api/rule.md [hunk 13]
```diff
@@ -1,27 +1,132 @@
...
+Designing your API up front can be very useful where you have a significant amount of backend development to complete before there is sufficient functionality for front end development to easily commence.
+
 <!--endintro-->
-Rather than waiting for the backend to be complete, the APIs can be agreed up front. Tools like **Azure API Management** then allow you to mock up the APIs with sample data, allowing the front end development to commence much sooner.  
- This can result in valuable feedback on the API and backend functionality much earlier in your development lifecycle. The key benefit is being able to present the frontend to stakeholders for feedback really early.  
 
+Rather than waiting for the backend to be complete, the APIs can be agreed on up front. Tools like Azure API Management then allow you to mock up the APIs with sample data, allowing the front end development to commence much sooner.
+
+This can result in valuable feedback on the API and backend functionality much earlier in your development lifecycle. The key benefit is being able to present the frontend to stakeholders for feedback really early.
```
[the remark this answers is not in the document]
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md:27]: done

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 2 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md]: done

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 3 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md]: done

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 12 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md:55]: Display Name = a nice friendly name for the operation that will be shown in the API Operations list and Dev Portal
Name = programmatic name that will be used to identify the operation in the swagger/openapi spec or when writing policy code

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 4 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md]: done

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 5 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md]: done

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 6 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md]: done

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 7 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md]: done

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 8 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md]: done

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 9 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md]: done

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 11 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md]: done

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 10 above)
Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md]: done

**Review by bradystroud — APPROVED** [review]

File: rules/do-you-know-when-to-mockup-your-api/rule.md (hunk 12 above)
Comment (pierssinclairssw) [reply rules/do-you-know-when-to-mockup-your-api/rule.md:55]: Sorry, to clarify I meant from the readers perspective. I am aware but maybe the reader of the rule would not be?

Comment (william-liebenberg) [reply rules/do-you-know-when-to-mockup-your-api/rule.md:55]: done

**Review by pierssinclairssw — APPROVED** [review]

**Pull request merged**